download.sh

```diff
@@ -0,0 +1,71 @@
+#! /usr/bin/env sh
+# Download the latest Wikipedia Enterprise dumps.
+# Exit codes:
+# - No new dumps available
+# - Dump not complete
+USAGE="download.sh DOWNLOAD_DIR"
+
+set -eu
+set -x
+
+if [ -z "${1}" ]
+then
+    printf "Usage:\t%s\n" "$USAGE" >&2
+    exit 1
+fi
+
+DOWNLOAD_DIR=$1
+
+
+# Write printf-style message to stderr with a timestamp and line ending.
+log () {
+    printf "%s " "$(date '+%Y-%m-%dT%H:%M:%SZ')" >&2
+    # shellcheck disable=2059 # format string is part of arguments
+    printf "$@" >&2
+    printf "\n" >&2
+}
+
+# Ensure we're running in the directory of this script.
+SCRIPT_PATH=$(dirname "$0")
+cd "$SCRIPT_PATH"
+SCRIPT_PATH=$(pwd)
+
+if [ -z "${LANGUAGES+}" ]
+then
+    LANGUAGES=$(jq -r '(.sections_to_remove | keys)' article_processing_config.json)
+fi
+log "Selected languages: %s" "$LANGUAGES"
+
+TMP=$(mktemp -d wikiparser-download-XXXX)
+trap 'rm -rf $TMP' EXIT INT HUP
+
+log "Fetching run index"
+# Call wget outside of pipeline for errors to be caught by set -e.
+wget 'https://dumps.wikimedia.org/other/enterprise_html/runs/' --no-verbose  -O "$TMP/runs.html"
+
+LATEST_DUMP=$(grep -Po '(?<=href=")[^"]*' "$TMP/runs.html" | sort | head -n1)
+LATEST_DUMP="${LATEST_DUMP#/}"
+
+log "Fetching index for latest dump '%s'" "$LATEST_DUMP"
+wget "https://dumps.wikimedia.org/other/enterprise_html/runs/$LATEST_DUMP" --no-verbose -O "$TMP/$LATEST_DUMP.html"
+
+for lang in $LANGUAGES
+do
+    url="https://wikipedia.invalid/${LATEST_DUMP}/${lang}wiki-NS0-${LATEST_DUMP}-ENTERPRISE-HTML.json.tar.gz"
+    if ! wget --no-verbose --method=HEAD "$url"
+    then
+        log "Dump for '%s' does not exist yet at '%s'" "$lang" "$url"
+        continue
+    fi
+    URLS="$URLS $url"
+done
+
+if [ -z "$URLS" ]
+then
+    log "No dumps available"
```

Comment:
If URLS is empty, then none of the specified languages could be found for the latest dump.

If a newer dump isn't available, it will still check the sizes of the last downloaded dump, and exit with 0.

Reply:
Good! The goal is to make a cron script that will update files automatically when they are published (and delete old files).

Another question: should previously generated HTML and other temporary files be deleted before relaunching the wikiparser? Does it make sense to cover it in the run script?

Reply:
They shouldn't _need_ to be.

The temporary files are regenerated each time.
The generated HTML will be overwritten if it is referenced in the new planet file.

If an article isn't extracted from the dump due to #24 or something else, then having the old copy still available might be useful.

But if the HTML simplification is changed, and older articles are no longer referenced in OSM, then they will remain on disk unchanged.